### ISSUE TITLE
Equation numbering not working on my files?

### ISSUE THREAD

**Issue by tonyriverms** [sphinx-doc/sphinx#5026]
I am relatively new to sphinx and I just created two files to test equation cross reference with numbering. 

I do the following configuration in conf.py

``` python
numfig = True
math_numfig = True
numfig_secnum_depth = 2
math_eqref_format = "Eq.{number}"
```

The first file is 

```
1. Test File 1 Main
=============

Inline math examples: :math:`\color{blue}{\sigma_{1}}` equals :math:`\colorbox{yellow}{\sigma_{2}}` then etc, etc.

Any text.

.. math::
 x^2+y^2=1
 :label: eq_a

Math block example with label:

.. math::
 e^{i\pi} + 1 = 0
 :label: eq_b


1.1 Some Examples
****************

.. math::
 \color{red}{x^2}+y^2=3
 :label: eq_c
```

The second one is 

```
2. Test File 2 Main
=============

Refer to :eq:`eq_a`

Refer to :eq:`eq_b`

Refer to :eq:`eq_c`
```

The output is HTML. It does not work for the numbering, all labels are missing (but links are good). It is somewhat similar to https://github.com/sphinx-doc/sphinx/pull/4166.

File 1 html output:
<img width="948" alt="capture" src="https://user-images.githubusercontent.com/9701455/40802915-fe896480-64e4-11e8-9233-f9de1b2e1d1c.PNG">

File 2 html output:
<img width="259" alt="capture2" src="https://user-images.githubusercontent.com/9701455/40802933-07a1393a-64e5-11e8-9654-59fbf52c4d08.PNG">

Am I missing something? I am also puzzled how to automatically number the section numbers. The section numbers in the test are manually added. Is it because I need to config section numbers in order for the equation number to appear? What is the right way to do so? Thanks!

**Comment by tk0miya** [comment]
Thank you for reporting.

But I missed your environment information. What version of Sphinx did you use? What type of math extension did you use?
I need these reproducible information to answer you.
Please post this again with our issue format.

Thanks,

**Comment by tk0miya** [comment]
Note: I can't reproduce this with Sphinx-1.7.5 and sphinx.ext.mathjax.

**Comment by tonyriverms** [comment]
Thank you for trying to help. **I did not add :numbered: in the toctree.** Although it looks trivial, but it did take me a while to figure out. We could improve this by adding a notice to the documentation about this (I am not sure if the documentation mentioned this somewhere, but I didn't see it).
